util/chplenv/chpl_gpu.py

```diff
@@ -15,6 +15,19 @@
     "rocm": ("CHPL_ROCM_PATH", "hipcc", 3,"gfx906")
 }
 
+def determineGpuType():
+    gpuTypesFound = 0
+    typeFound = None
+    for gpuType in GPU_TYPES.keys():
+        if which(GPU_TYPES[gpuType][1]):
+            typeFound = gpuType
+            gpuTypesFound += 1
+
+    if gpuTypesFound != 1:
+        error("Unable to determine GPU type, assign 'CHPL_GPU_CODEGEN' to one of {}".
+            format(GPU_TYPES.keys()))
+    return typeFound
```

Comment:
Nitpicky: you could use a `gpuTypes = []` array, push to it each time `which` succeeds, and have the error message report which supported types were found. Then, no need for a counter (you've got `len`!) and the error message is a bit friendlier.

Reply:
I'd keep "one of {}" was before, but maybe add a parenthetical ("detected bla, bla SDKs") like this.

Reply:
Sure, updated in 0493a4bdef99245b760c4d2929a553ef2eeebeb1 to add a parenthetical if any gpuTypes were detected.